core/lib/server/pub.ts

```diff
@@ -576,29 +581,35 @@ export const createPubRecursiveNew = async <Body extends CreatePubRequestBodyWit
 				stageId: createdStageId ?? null,
 				values: hydratedValues,
 				children: children.length ? children : [],
+				depth,
```

Comment:
I'm passing the depth around everywhere (probably in some places it doesn't totally make sense) to satisfy the type, which I made include the depth. There is probably a more correct way to handle this!